__tests__/app.test.js

```diff
@@ -47,3 +47,66 @@ describe('GET /api/topics', () => {
             });
     });
 });
+describe('GET /api', () => {
+    test('200: should return an object containing the description for this endpoint', () => {
+        return request(app)
+            .get('/api')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.endpoints).toBe('object');
+                expect(Object.keys(body.endpoints['GET /api'])).toContain('description');
+                expect(typeof body.endpoints['GET /api'].description).toBe('string');
+            });
+    });
+    test('200: should return an object containing the description, valid queries and example response for /api/topics endpoint', () => {
+        return request(app)
+            .get('/api')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.endpoints).toBe('object');
+                expect(Object.keys(body.endpoints['GET /api/topics'])).toContain('description', 'queries', 'exampleResponse');
+                expect(typeof body.endpoints['GET /api/topics'].description).toBe('string');
+                expect(Array.isArray(body.endpoints['GET /api/topics'].queries)).toBe(true);
+                expect(body.endpoints['GET /api/topics'].queries.length).toBe(0);
+            });
+    });
+});
+describe('GET /api/articles/:article_id', () => {
+    test('200: should return an article object containing the expected fields', () => {
+        return request(app)
+            .get('/api/articles/1')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Object.keys(body.article)).toContain('author', 'title', 'article_id', 'body', 'topic', 'created_at', 'votes', 'article_img_url');
+            });
+    });    
+    test('200: should return an article object containing data from the test dataset', () => {
+        return request(app)
+            .get('/api/articles/2')
+            .expect(200)
+            .then(({body}) => {
+                expect(typeof body.article).toBe('object');
+                expect(Object.keys(body.article)).toContain('author', 'title', 'article_id', 'body', 'topic', 'created_at', 'votes', 'article_img_url');
+                Array('author', 'title', 'body', 'topic', 'article_img_url').forEach((keyName) => {
+                    expect(body.article[keyName]).toBe(data.articleData[1][keyName]);
+                })
+            });
```

Comment:
This is a little difficult to read. I would recommend refactoring this so its easier to follow and understand what is actually happening. I would recommend looking into expect.objectContaining or .toMatchObject for this.